### PR TITLE
Use i-figured for section-based equation numbering

### PR DESCRIPTION
**Pull request by JoshuaLampert**
So currently, the equation numbering and labeling isn't great. Personally, I like the following setup the most:
- Usual one-line equations get the number _(ChapterNumber.EquationNumberWithinChapter)_ and they are also referenced like this (i.e. without supplement and with parentheses). They should only be numbered if they have a label.
- For multi-line equations, we have the pattern _(ChapterNumber.EquationNumberWithinChapter.SubEquationNumber)_ and again, they are referenced exactly like this and only subequations with a label get a number. It should also be possible to reference the whole equation as _(ChapterNumber.EquationNumberWithinChapter)_.

Unfortunately, I think this setup is currently (not easily) possible. [`equate`](https://github.com/EpicEricEE/typst-equate) doesn't work well for section-based numbering, cf. https://github.com/EpicEricEE/typst-equate/issues/3 and [`i-figured`](https://github.com/RubixDev/typst-i-figured) cannot handle the subequations.

Therefore, I would go for the following workaround, which allows two different settings
1. If `equate-settings` is `none` (default), we apply section-based numbering using `i-figured`, but loose the possibility to have individual numbers for subequations (i.e. multi-line equations will only get one number, which can be referenced). In case we will always have the pattern _(ChapterNumber.EquationNumberWithinChapter)_. `i-figured` allows to number only labeled equations. Note that in this case, the reference needs an additional `eq:` (how this prefix looks like is configurable) prepended to the actual label.
2. If `equate-settings` are passed, we use `equate`, but loose the possibility for section-based equation numbering. ~In this case, one should pass `sub-numbering: true` to the settings (otherwise references to subequations will be wrong)~. This means, we will have (in both cases, only labeled equations get a number if `number-mode: "label"` in the settings)
	- the pattern _(EquationNumberWithinDocument)_ for single-line equations
	- the pattern _(EquationNumberWithinDocument.SubEquationNumber)_ for subequations in multi-line equations. The whole equation can still be referenced as _(EquationNumberWithinDocument)_.

### CODE REVIEW

File: template/main.typ [hunk 1]
```diff
@@ -30,7 +30,6 @@
 
   // equation settings
   equate-settings: (breakable: true, sub-numbering: true, number-mode: "label"),
-  numbering-pattern: "(1.1)",
```
Comment (sebaseb98) [inline template/main.typ:33]:
```suggestion
	equation-numbering-pattern: "(1.1)",
	
```
I would like to have all template arguments in the main for now (this is also something we can discuss on)

File: lib.typ [hunk 2]
```diff
@@ -51,28 +52,56 @@ show ref: set text(fill: link-color)
...
-  } else if it.has("label"){
-    math.equation(block:true, numbering: numbering-pattern, it)
   } else {
     it
   }
 }
+set math.equation(numbering: equation-numbering-pattern) if equate-settings != none
+
+// Reference equations with parentheses (for equate)
+// cf. https://forum.typst.app/t/how-can-i-set-numbering-for-sub-equations/1603/4
```
Comment (sebaseb98) [inline lib.typ:77]: Love it, that was something I also tried by myself but couldn't make it work yet!

**Review by sebaseb98 — CHANGES_REQUESTED** [review]
Thanks for all the effort, I also really like the idea of having both options of chapter based equation numberings and subequation numberings, but I think the user experience is not optimal. 

I would love to be able to shift between the options by just changing the template parameters, but currently the `i-figured` prefixes don't allow for this...

**Comment by JoshuaLampert** [comment]
> I would love to be able to shift between the options by just changing the template parameters, but currently the i-figured prefixes don't allow for this...

Yes, I totally agree. I am also not happy with that. But I don't see a solution for that right now. Unfortunately, i-figured needs this prefix.

**Review by sebaseb98 — APPROVED** [review]

**Pull request merged**